Framework/API/inc/MantidAPI/InfoComponentVisitor.h

```diff
@@ -64,9 +71,19 @@ class MANTID_API_DLL InfoComponentVisitor
   /// Only Assemblies and other NON-detectors yield ranges
   std::vector<std::pair<size_t, size_t>> m_ranges;
 
+  /// Positions
+  boost::shared_ptr<std::vector<Eigen::Vector3d>> m_positions;
+
+  /// Rotations
+  boost::shared_ptr<std::vector<Eigen::Quaterniond>> m_rotations;
```

Comment:
Ultimately we want `ComponentInfo` to store positions/rotations as `cow_ptr`. Creating as `shared_ptr` allow us to pass the same objects over from the `InfoComponentVisitor` to the `ComponentInfo`. When the temporary `InfoComponentVisitor` goes out of scope, the `ComponentInfo` becomes the sole owner (at least until copies etc are made).